Makefile

```diff
@@ -36,31 +36,43 @@ node_modules: package.json yarn.lock ## Install node modules.
 
 ##@ Swagger
 SPEC_TARGET = public/api-spec.json
-MERGED_SPEC_TARGET := public/api-merged.json
+ENTERPRISE_SPEC_TARGET = public/api-enterprise-spec.json
+MERGED_SPEC_TARGET = public/api-merged.json
 NGALERT_SPEC_TARGET = pkg/services/ngalert/api/tooling/api.json
 
 $(NGALERT_SPEC_TARGET):
 	+$(MAKE) -C pkg/services/ngalert/api/tooling api.json
 
-$(MERGED_SPEC_TARGET): $(SPEC_TARGET) $(NGALERT_SPEC_TARGET) $(SWAGGER) ## Merge generated and ngalert API specs
+$(MERGED_SPEC_TARGET): swagger-oss-gen swagger-enterprise-gen $(NGALERT_SPEC_TARGET) $(SWAGGER) ## Merge generated and ngalert API specs
 	# known conflicts DsPermissionType, AddApiKeyCommand, Json, Duration (identical models referenced by both specs)
-	$(SWAGGER) mixin $(SPEC_TARGET) $(NGALERT_SPEC_TARGET) --ignore-conflicts -o $(MERGED_SPEC_TARGET)
+	$(SWAGGER) mixin $(SPEC_TARGET) $(ENTERPRISE_SPEC_TARGET) $(NGALERT_SPEC_TARGET) --ignore-conflicts -o $(MERGED_SPEC_TARGET)
 
-$(SPEC_TARGET): $(SWAGGER) ## Generate API Swagger specification
+swagger-oss-gen: $(SWAGGER) ## Generate API Swagger specification
+	@echo "re-generating swagger for OSS"
+	rm -f $(SPEC_TARGET)
 	SWAGGER_GENERATE_EXTENSION=false $(SWAGGER) generate spec -m -w pkg/server -o $(SPEC_TARGET) \
 	-x "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions" \
 	-x "github.com/prometheus/alertmanager" \
 	-i pkg/api/swagger_tags.json \
-	--exclude-tag=alpha
-	go run pkg/services/ngalert/api/tooling/cmd/clean-swagger/main.go -if $@ -of $@
```

Comment:
Should we be deleting the alert spec gen? On the other hand I'm also not sure why [this file](https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/api/tooling/cmd/clean-swagger/main.go) generates the `"Unified Alerting API"` spec separately. 🤔 @papagian do you have more context?

Reply:
I think we can safely remove this line given that [it's already executed](https://github.com/grafana/grafana/blob/f986adee362b9d72d890acf3bc0135563182f195/pkg/services/ngalert/api/tooling/Makefile#L31) as part of [this target](https://github.com/grafana/grafana/blob/f986adee362b9d72d890acf3bc0135563182f195/Makefile#L43)


> On the other hand I'm also not sure why [this file](https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/api/tooling/cmd/clean-swagger/main.go) generates the `"Unified Alerting API"` spec separately 🤔 but we can keep it this way since It Works ™️ ! @papagian do you have more context?

the alerting swagger was introduced before the general one
also, its swagger annotations are used for both creating the alerting spec but also for creating the [server stubs](https://github.com/grafana/grafana/blob/f986adee362b9d72d890acf3bc0135563182f195/pkg/services/ngalert/api/tooling/Makefile#L39)
we could consider changing the server generation and use the general swagger and targeting only the alerting operations by setting the `--tags` appropriately
but I think that it doesn't worth the effort (at least for now)

Reply:
thank you for this context! agreed that we can leave it as is for now :)